wgsl/index.bs

```diff
@@ -1697,6 +1700,327 @@ Issue: (dneto): Do we have to explicitly list the type environment Gamma? That's
+  <caption>Binary arithmetic expressions over vectors</caption>
+  <thead>
+    <tr><td>Precondition<td>Conclusion<td>Notes
+  </thead>
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec*<td class="nowrap">`e1 + e2` : *T*<td>Component-wise integer addition (OpIAdd)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *FloatVec*<td class="nowrap">`e1 + e2` : *T*<td>Component-wise floating point addition (OpIAdd)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec*<td class="nowrap">`e1 - e2` : *T*<td>Component-wise integer subtraction (OpISub)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *FloatVec*<td>`e1 - e2` : *T*<td>Component-wise floating point subtraction (OpISub)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec*<td>`e1 * e2` : *T*<td>Component-wise integer multiplication (OpIMul)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *FloatVec*<td>`e1 * e2` : *T*<td>Component-wise floating point multiplication (OpIMul)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec* with unsigned component<td>`e1 / e2` : *T*<td>Component-wise unsigned integer division (OpUDiv)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec* with signed component<td>`e1 / e2` : *T*<td>Component-wise signed integer division (OpSDiv)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *FloatVec*<td>`e1 / e2` : *T*<td>Component-wise floating point division (OpFDiv)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec* with unsigned component<td>`e1 % e2` : *T*<td>Component-wise unsigned integer modulus (OpUMod)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *IntVec* with signed component<td>`e1 % e2` : *T*<td>Component-wise signed integer remainder (OpSMod)
+  <tr><td>*e1* : *T*<br> *e2* : *T*<br> *T* is *FloatVec*<td>`e1 % e2` : *T*<td>Component-wise floating point division (OpFMod)
+</table>
+
+<table class='data'>
+  <caption>Binary arithmetic expressions with mixed scalar, vector, and matrix operands</caption>
+  <thead>
+    <tr><td>Precondition<td>Conclusion<td>Notes
+  </thead>
+  <tr><td>*e1* : f32<br>
+          *e2* : *T*<br>
+          *T* is *FloatVec*
+      <td>`e1 * e2` : *T*<br>
+          `e2 * e1` : *T*
+      <td>Scalar multiplication of vector (OpVectorTimesScalar)
+  <tr><td>*e1* : f32<br>
+          *e2* : *T*<br>
+          *T* is mat*N*x*M*&lt;f32&gt;
+      <td>`e1 * e2` : *T*<br>
+          `e2 * e1` : *T*
+      <td>Scalar multiplication of matrix (OpMatrixTimesScalar)
+  <tr><td>*e1* : vec*M*&lt;f32&gt;<br>
+          *e2* : mat*N*x*M*&lt;f32&gt;
+      <td>`e1 * e2` : vec*N*&lt;f32&gt;<br>
+      <td>Vector times matrix (OpVectorTimesMatrix)
+  <tr><td>*e1* : mat*N*x*M*&lt;f32&gt;<br>
+          *e2* : vec*N*&lt;f32&gt;
+      <td>`e1 * e2` : vec*M*&lt;f32&gt;<br>
+      <td>Matrix times vector (OpMatrixTimesVector)
+  <tr><td>*e1* : mat*K*x*N*&lt;f32&gt;<br>
+          *e2* : mat*M*x*K*&lt;f32&gt;<br>
+      <td>`e1 * e2` : mat*M*x*N*&lt;f32&gt;<br>
+      <td>Matrix times matrix (OpMatrixTimesMatrix)
+</table>
```

Comment:
I don't see matrix x matrix multiplication in here. I expect we follow the same rules we learned in trigonometry class, but it might be worth mentioning for completeness.

Reply:
?  Matrix times matrix is at line 1773-1776